integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/segment/ShowSegmentTestCase.scala

```diff
@@ -0,0 +1,207 @@
+    sql("drop table if exists source")
+    sql(
+      """
+        |create table source (age int)
+        |STORED AS carbondata
+        |partitioned by (name string, class string)
+        |TBLPROPERTIES('AUTO_LOAD_MERGE'='true','COMPACTION_LEVEL_THRESHOLD'='2,2')
+        |""".stripMargin)
+    sql("insert into source select 1, 'abc1', 'classA'")
+    sql("insert into source select 2, 'abc2', 'classB'")
+    sql("insert into source select 3, 'abc3', 'classA'")
+    sql("insert into source select 4, 'abc4', 'classB'")
+    sql("insert into source select 5, 'abc5', 'classA'")
+    sql("insert into source select 6, 'abc6', 'classC'")
+    sql("show segments on source").show(false)
+
+    val df = sql(s"""show segments on source""").collect()
+    // validating headers
+    val header = df(0).schema
+    assert(header(0).name.equalsIgnoreCase("ID"))
+    assert(header(1).name.equalsIgnoreCase("Status"))
+    assert(header(2).name.equalsIgnoreCase("Load Start Time"))
+    assert(header(3).name.equalsIgnoreCase("Spent"))
+    assert(header(4).name.equalsIgnoreCase("Partition"))
+    assert(header(5).name.equalsIgnoreCase("Data Size"))
+    assert(header(6).name.equalsIgnoreCase("Index Size"))
+    val col = df
+      .map(row => Row(row.getString(0), row.getString(1)))
+      .filter(_.getString(1).equals("Success"))
+      .toSeq
+    assert(col.equals(Seq(Row("4.1", "Success"), Row("0.2", "Success"))))
+
+    sql(
+      """
+        | show segments on source as
+        | select id, status, datasize from source_segments where status = 'Success' order by dataSize
+        |""".stripMargin).show(false)
+
+    val rows = sql(
+      """
+        | show segments on source as
+        | select id, status, datasize from source_segments where status = 'Success' order by dataSize
+        |""".stripMargin).collect()
+
+    assertResult("4.1")(rows(0).get(0))
+    assertResult("Success")(rows(0).get(1))
+    assertResult(1762)(rows(0).get(2))
+    assertResult("0.2")(rows(1).get(0))
+    assertResult("Success")(rows(1).get(1))
+    assertResult(3524)(rows(1).get(2))
+
+    val tables = sql("show tables").collect()
+    assert(!tables.toSeq.exists(_.get(1).equals("source_segments")))
+
+    sql(s"""drop table source""").collect
+  }
+
+  test("Show Segments on empty table") {
+    sql(s"""drop TABLE if exists source""").collect
+    sql(s"""CREATE TABLE source (CUST_ID int,CUST_NAME String,ACTIVE_EMUI_VERSION string,DOB timestamp, DOJ timestamp, BIGINT_COLUMN1 bigint,BIGINT_COLUMN2 bigint,DECIMAL_COLUMN1 decimal(30,10),Double_COLUMN1 double,DECIMAL_COLUMN2 decimal(36,10), Double_COLUMN2 double,INTEGER_COLUMN1 int) STORED AS carbondata TBLPROPERTIES('table_blocksize'='1')""").collect
+    checkAnswer(sql("show segments on source"), Seq.empty)
+    val result = sql("show segments on source as select * from source_segments").collect()
+    assertResult(0)(result.length)
+  }
+
+  test("can not show segments on exist table") {
```

Comment:
change the test name to `test show segments on already existing table`

Reply:
fixed